### ISSUE TITLE
Duplicate frames when transcoding a DVD using detelecine

### ISSUE THREAD

**Issue by loshlee** [donmelton/more-video-transcoding#7]
I bought a DVD for a TV show that I really liked, and I'm transcoding it for my Apple Family Sharing library.

 I have already ripped it, already matched and concatenated the VOBs to mpegs to get the playback structure I want, but now I'm stumped upon trying to transcode.
 
I have a dilemma. After a lot of trial and error, I find that (with the options I'm aware of) I can have either ideal picture quality but duplicate frames, or I can have decent picture with no duplicate frames.

I can't tolerate duplicate frames. To me, it's like a slap in the face whenever one passes.
 
 I've tried two-pass-transcode.rb successfully with the following options, except that they yield the occasional duplicate frame.
 
-x detelecine -x deinterlace=skip-spatial -x rate=23.976 -x hqdn3d=light -x cfr 

I've tried ffmpeg using the following options, which make it slow, yielding less than ideal picture quality, but no duplicate frames.

-vf 'pullup,bwdif,hqdn3d=2:1:2' -r 24000/1001 -c:v libx264 -tune film -crf 14 -preset slower

I've also tried the "fieldmatch" filter with ffmpeg as included with Other Video Transcoding, which also yields really good pictures, but also leaves some duplicate frames.

Can you recommend a way to tweak the two-pass-transcode.rb options I'm using to eliminate the duplicate frames? Or please suggest a different approach. Thanks.

L. Lee

**Comment by skj-dev** [comment]
Unless you're really constrained for space, I would suggest not transcoding the DVD content in the first place. Personally, I don't transcode anything that's less than 1080p. 

You might be better served by ripping the DVD into its episodes with MakeMKV instead of extracting the VOBs. At least then you'd have access to other tools, like the MKVToolNix collection, to manipulate the files, such as concatenating them. 

This is outside my wheelhouse, since I tend to avoid DVD formatted content, but you might be getting caught by different software trying to adjust for different frame rates. That might be where the duplicate frames are coming from. I don't know if ripping via MakeMKV would alleviate any of that, if that's the problem. 

Just my 2c. :)

**Comment by loshlee** [comment]
> Unless you're really constrained for space, I would suggest not transcoding the DVD content in the first place. Personally, I don't transcode anything that's less than 1080p.
> 
> You might be better served by ripping the DVD into its episodes with MakeMKV instead of extracting the VOBs. At least then you'd have access to other tools, like the MKVToolNix collection, to manipulate the files, such as concatenating them.

Thanks! Yes, I was using MakeMKV's title extracts as source for transcoding, but maybe I'll just take 'em as they are. The content is available online in HD format, but not without ads, which are at least as unwanted as dup frames. (That's why I made the DVD purchase.) Maybe I'll take a stab at removing them.

> Just my 2c. :)

Always appreciated!

L. Lee

**Comment by lisamelton** [comment]
@loshlee While I would definitely agree with @ttyS0 that you're probably better off *not* transcoding DVD content, there is a simpler formula that usually works better if you *want* to do that.

The key is to not use the `deinterlace` filter along with the `detelecine` filter. Just use `detelecine`. Also, it's better to not pass it any arguments. IOW, you shouldn't need to specify `skip-spatial`.

Also, the `hqdn3d` filter is overkill. It's finicky and can actually cause more problems than it supposedly cures. I recommend avoiding it.

Here's what I use for most NTSC DVDs: `-x detelecine -x rate=23.976 -x cfr`. And that's it. 🤷‍♀️

If that doesn't looks good to you, I would definitely not bother transcoding them.

I hope that helps.

**Comment by loshlee** [comment]
Thanks, Don.

That's what I started with:
`-x detelecine -x rate=23.976 -x cfr`

It was good, except that it didn't provide decombing, which really is a real deal breaker for this video. I can deinterlace without detelecine, but only at the expense of natural motion. This DVD has both interlacing and pulldown. I have 3 progressive frames and 2 interlaced frames, and that pattern is absolutely consistent. When I detelecine, and make the frame rate 23.976, there's a ghost for that combined, deinterlaced frame which consists of the sharp horizontal lines. I haven't found a way to get rid of those lines, but I can mitigate them using yadif or bwdif, and further obscure them with hqdn3d. That's what prevents the duplicate frames from being completely removed, I think. Is there any other way to deal with such video? I wouldn't think that's too uncommon. Maybe there's a method I've missed, but I'll tell you, it's a very winding path that led me to post the question here.

Can't believe I originally typed "24.976".
Thanks again.

L. Lee

**Comment by loshlee** [comment]
One more thing, I will not leave it at 29.97 and get slapped in the face 6 times a second when there is smooth panning. I can take a few lines better than that.

Yours 'til I don't have to,

L. Lee

**Comment by lisamelton** [comment]
@loshlee It's almost like interlace DVDs were invented to confound us all. 🤷‍♀️

Do you want us all to admit defeat and close this now?

**Comment by loshlee** [comment]
OK.

**Comment by loshlee** [comment]
I'm not reopening this issue, but I wanted to tell you that I've discovered that in this particular case, there's a chance that the detelecine can be accomplished through deinterlacing and forcing frame rate to 23.976. I doubt if that's a technically correct description, but that's how it looks to me. Please see [this](https://www.reddit.com/r/shutterencoder/comments/1384pcr/hoping_to_try_shutter_encoder_for_my_dvd/).

**Comment by lisamelton** [comment]
@loshlee Thanks for the link! Yeah, that *can* work in some cases. The problem is that forcing the frame rate without detelecining can cause some "jumpy" sequences. As always, YMMV.

**Comment by loshlee** [comment]
Still not reopening, but I wanted to contribute my solution. Using the help I got for [Shutter Encoder](https://www.shutterencoder.com/), I was able to formulate methods that let me transcode for Apple Home Sharing so that the output looks very close to exactly like the input. The interesting thing is that, depending on what passage I use, one or the other of these ffmpeg command lines yields desirable result. Fortunately, when I halt the process soon after it starts, the integrity of the output file will be completed by ffmpeg so that it's viable. Doing that allows me to test a very short passage before committing to one or the other of the commands.

Must be something about the start of the source file that determines which method works, because if I create an excerpt from a video, that excerpt may require either of the two methods in turn. I don't know enough details to make the specification when cutting. It's always a toss-up.

`ffmpeg  -i "DVD_Title.mpeg" -r 24000/1001 -c:v libx264 -crf 11 -preset veryslow -level 5.1 -tune film -vf "pullup,yadif=0,hqdn3d=4:4:4:4" -c:a copy -pix_fmt yuv420p -sws_flags bicubic -movflags faststart "DVD_Title.mp4"`

`ffmpeg -i "DVD_Title.mpeg" -r 24000/1001 -c:v libx264 -crf 11 -preset veryslow -level 5.1 -tune film -vf "detelecine=first_field=bottom,hqdn3d=4:4:4:4" -c:a copy -pix_fmt yuv420p -sws_flags bicubic -movflags faststart  "DVD_Title.mp4"`
